scripts/apps/search/directives/SearchContainer.ts

```diff
@@ -28,6 +36,18 @@ export function SearchContainer() {
                         }
                     }
                 });
+                preferencesService.get().then((result) => {
+                    if (result !== undefined && FILTERS_PANEL_PREFERENCES_KEY in result) {
+                        this.flags.facets = result[FILTERS_PANEL_PREFERENCES_KEY]['enabled'];
+                    } else {
+                        this.flags.facets = filtersPanelPreferences[FILTERS_PANEL_PREFERENCES_KEY]['default'];
+                    }
+                });
+                $scope.toggleFiltersPane = () => {
+                    filtersPanelPreferences[FILTERS_PANEL_PREFERENCES_KEY]['enabled'] = !$scope.flags.facets;
+                    preferencesService.update(filtersPanelPreferences);
+                    $scope.flags.facets = !$scope.flags.facets;
```

Comment:
it looks like we're saving the boolean value in two places

1. `filtersPanelPreferences[FILTERS_PANEL_PREFERENCES_KEY]['enabled']`
2. `$scope.flags.facets`

is this necessary?

Reply:
yes, one is for scope another is for "payload". I've upadted that part a little bit.